shiny/workflowPlot/server.R

```diff
@@ -122,11 +159,38 @@ server <- shinyServer(function(input, output, session) {
                plt <- plt + geom_line()
              }
       )
-      # geom_point() +
+    plt <- plt + labs(title=title, x=xlab, y=ylab) + geom_smooth(n=input$smooth_n)
+    # Check if user wants to load external data
+    # Similar to using event reactive
+    if (input$load_data>0) { 
+      File_format <- getFileFormat(bety,input$formatID)
+      # Retaining the code for getting file format using inputRecordID
+      # File_format <- getFileFormat(bety,input$inputRecordID)
+      ids_DF <- parse_ids_from_input_runID(input$all_run_id)
+      settings <- getSettingsFromWorkflowId(bety,ids_DF$wID[1])
+      inFile <- input$fileUploaded
+      externalData <- loadObservationData(bety,settings,inFile$datapath,File_format)
+      # If variable found in the uploaded file 
+      if (input$variable_name %in% names(externalData)){
+        externalData <- externalData %>% dplyr::select(posix,dplyr::one_of(input$variable_name))
```

Comment:
This code is missing a call to PEcAn.benchmark::align_data. That call needs to be done BEFORE the plotting of the model output, because it may change the timestep/averaging, start/end date, and time zone of the model output. This will also facilitate the more general use of existing Metric functions in the plotting, allowing users to easily generate a set of 'standard' plots and scores.